### PR TITLE
#174797525 Multi city trips

### PR DESCRIPTION
**Pull request by elvisiraguha**
#### What does this PR do?
create a multi-city trips request endpoint
#### Description of Task to be completed?
- create a route for multi-city trips
- create a controller for multi-city trips
- add validations for multi-city router
- create a table for trips
- create a table for requests
- add tests for multi-city endpoint
- add swagger documentation
#### How should this be manually tested?
- Clone the repo git clone `https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
- Switch your working directory `cd script-heroes-bn-backend`
- Checkout the branch git checkout `ft-multi-city-request-#174797525`
- Reference _.env-example_ and setup all variables in _.env_ file
- Install all dependencies `npm install`
- Create migrations `npm run migrate`
- Create seeds `npm run seed`
- Start server `npm run dev`
- Send a GET request to _localhost:3000/api/locations_ to get the available locations with their IDs
- Send a POST request to _localhost:3000/api/trips/multi-city_ to test this endpoint
#### Any background context you want to provide?
Example request:
```
{
    "trips": [
        {
            "origin": 1,
            "destination": 3,
            "from": "2020-10-02",
            "till": "2020-10-04"
        },
        {
            "origin": 3,
            "destination": 5,
            "from": "2020-10-04",
            "till": "2020-10-06"
        },
        {
            "origin": 5,
            "destination": 4,
            "from": "2020-10-06",
            "till": "2020-10-08"
        },
        {
            "origin": 4,
            "destination": 2,
            "from": "2020-10-08",
            "till": "2020-10-09"
        }
    ]
}
```
Preset locations: 
```
[
        {
          city: 'Kigali',
          country: 'Rwanda'
        },
        {
          city: 'Nairobi',
          country: 'Kenya'
        },
        {
          city: 'Pretoria',
          country: 'South Africa'
        },
        {
          city: 'Istanbul',
          country: 'Turkey'
        },
        {
          city: 'London',
          country: 'United Kingdom'
        }
]
```
Preset request types:
```
[
        {
          type: 'one-way'
        },
        {
          type: 'return'
        },
        {
          type: 'multi-city'
        }
]
```
#### What are the relevant pivotal tracker stories?
[#174797525](https://www.pivotaltracker.com/story/show/174797525)
#### Screenshots (if appropriate)
Screen Shot 2020-10-08 at 16 27 59

### CODE REVIEW

**Review by dushimeemma — APPROVED** [review]
Good job

**Review by Habinezajanvier — APPROVED** [review]
good

File: src/controllers/emailAuth.js [hunk 1]
```diff
@@ -36,22 +34,30 @@ class UserController {
         res.status(201).json({ message: res.__('Successfully registered') });
       })
       .catch((error) => {
-        res.status(500).send({ message: res.__('Something went wrong') });
+        res
+          .status(500)
+          .send({ message: res.__('Something went wrong'), error });
```
Comment (kipropbrian) [inline src/controllers/emailAuth.js]: should be res.json

File: src/database/migrations/20201013100651-update-accesstoken.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.changeColumn('AccessTokens', 'token', {
+      type: Sequelize.STRING(1234)
+    });
+  },
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.removeColumn('AccessTokens', 'token');
```
Comment (kipropbrian) [inline src/database/migrations/20201013100651-update-accesstoken.js]: The reverse should be to return it to STRING(255)

**Review by valens10 — APPROVED** [review]

**Review by Adadie — COMMENTED** [review]
Is it supposed to return a null requestId? Thought the requestId is supposed to be in the Trips table
Screen Shot 2020-10-16 at 07 15 09

**Review by b0nbon1 — APPROVED** [review]

**Comment by elvisiraguha** [comment]
> Is it supposed to return a null requestId? Thought the requestId is supposed to be in the Trips table
> Screen Shot 2020-10-16 at 07 15 09

Was this the response? It shouldn't be like this, make sure you follow the instructions carefully.

**Review by Adadie — APPROVED** [review]

File: locales/en.json [hunk 3]
```diff
@@ -73,6 +73,74 @@
...
+	"At least one room is required": "At least one room is required",
+	"Please provide a valid url": "Please provide a valid url",
+	"Provide image for your accomodation": "Provide image for your accomodation",
+	"Image is required": "Image is required",
+	"Accomodation created successfully": "Accomodation created successfully",
+	"Room in facility name created successfuly": "Room in facility name created successfuly",
+	"Accomodation successfully updated": "Accomodation successfully updated",
+	"Room Deleted successfully": "Room Deleted successfully",
+	"Accomodation deleted successfully": "Accomodation deleted successfully"
+}
```
Comment (kipropbrian) [inline locales/en.json]: Add a newline and check the indentation on this file

File: src/routes/index.js [hunk 4]
```diff
@@ -2,12 +2,18 @@ import express from 'express';
...
+import requests from './requests';
+import location from './locations';
+import asyncHandler from '../middlewares/asynchandler';
+import RequestsController from '../controllers/requests';
 
 const router = express.Router();
 
 router.use('/auth', users);
 router.use('/accommodations', accomodation);
+router.use('/requests', requests);
```
Comment (kipropbrian) [inline src/routes/index.js]: Is it /trips or /requests

Comment (elvisiraguha) [reply src/routes/index.js]: fixed

**Review by kipropbrian — APPROVED** [review]
Looks good

**Pull request merged**